### ISSUE TITLE
ReactMapboxGl is not a function

### ISSUE THREAD

**Issue by Akshay-Choudhary** [alex3165/react-mapbox-gl#555]
I am trying to use the react-mapbox-gl library using the ES5 way as I want to render it in IE11. Followed all steps as needed for ES5 but I get a ReactMapboxGl is not a function error on the UI everytime I use the ES5 code.

FYI: I have added the ReactMapboxGl function inside the render method. If that is what causing the issue, let me know where should I place this function in my React code.

**Comment by Rob-ot** [comment]
Hey Akshay, I'm happy to provide assistance but I'm not sure what the issue is from your description. Is it possible for you to post the code?

**Comment by Akshay-Choudhary** [comment]
Sorry I won't be able to post my code but I can describe my issue more clearly.

So I want to make my code compatible to work on IE11 so in order to have that I imported the ReactMapboxGl from the 'react-mapbox-gl' using the following syntax `const ReactMapboxGl = require("react-mapbox-gl");` instead of `import ReactMapboxGl from 'react-mapbox-gl'`and then used the Marker and ZoomControl class inside ReactMapboxGl using this syntax: 
`const Marker = ReactMapboxGl.Marker;`
`const ZoomControl = ReactMapboxGl.ZoomControl;`.

The issue I was facing was with the ReactMapboxGl function inside my render method which was giving me this error `ReactMapboxGl is not a function`.

Here is a copy of the code that replicated exactly my code [https://github.com/alex3165/react-mapbox-gl](https://github.com/alex3165/react-mapbox-gl)  

Here is the piece of code that was giving me the error 
`const Map = ReactMapboxGl({
  accessToken: "someTokenKey"
});`

But luckily here is a solution that I found to my own problem:
`const Map = ReactMapboxGl.Map({
  accessToken: "someTokenKey"
});` Using the map function works for making it ES5 compatible as well.
